### DISCUSSION TITLE
CodeBuddies Project Updates (August 31st, 2020)

### DISCUSSION THREAD

**Discussion by lpatmo** [codebuddies/frontend#145] [General]
A few recent happenings (that I can remember)... please comment below if you'd like to give shout-outs I missed or point out other things I've forgotten!

## Front-end
- [ ] POST a form -- @bkbuilt has been working on this; @lpatmo still needs to follow up to help unblock and help w/ form validation
- [ ] Search feature -- almost merged; @lpatmo paired with @watchtheblur on the implementation; thanks to @angelocordon for reviewing! imo PR can be merged in after a couple of small edits.
- [x] Resource page -- congrats @bengineerdavis for the PR and thanks @angelocordon for reviewing and for the React Query integration
- [ ] Update FE contributing.md instructions explaining how to fork, keep branches updated, and make a PR -- thanks @peoray for picking this one up
- [ ] Refactor /resources page to use React Query -- thanks @peoray for picking up!
- [x] @tgrrr shared about https://github.com/tgrrr/backstitch, although in #proj-cbv3-frontend we discussed how we're not sure it'll be something we want to own yet
- [x] We updated our project boards at github.com/codebuddies/frontend/projects

Other important frontend MVP work (unassigned right now): 
- [ ] Pagination for /resources (estimate: easy)
- [ ] Edit a resource (estimate: harder; should only be worked on after POST above is complete)

## Back-end
- [ ] @watchtheblur and @lpatmo have been walking through PRs to create a new sandbox `/projects` endpoints following [this tracking issue](https://github.com/codebuddies/backend/issues/79) @BethanyG created. These sessions are public on codebuddies.org/hangouts, but most have not been recorded/streamed. We plan to stream more starting this week, though.
- [x] New PR template (merged!) -- @lpatmo and @BethanyG worked on this
- [ ] New Issue template (still being discussed, although maybe at the point where one of us can create a draft PR whenever we get a chance) 
- [ ] @nuageklow is working on cleaning up the three separate places we have contributing.md instructions -- reworking the outline on the README and updating contributing.md itself with the new instructions!
- [ ] Update to Postman to include a /search field -- @watchtheblur worked on this; @lpatmo reviewed and needs to review again now that requested changes have been made
- [ ] Tags discussion started by @BethanyG to raise up implementation alternative approaches and questions: https://github.com/codebuddies/backend/discussions/154
- [ ] Email address validation implementation -- some complicated work currently assigned to @BethanyG 
- [x] Thanks to @BethanyG for creating a tracking issue (and multiple sub-issues) for the /hangouts endpoint: https://github.com/codebuddies/backend/issues/160
- [x] We updated our project boards at github.com/codebuddies/backend/projects

Full list of "P1 for MVP" issues (thanks to @BethanyG) for labeling: https://github.com/codebuddies/backend/issues?q=is%3Aissue+is%3Aopen+label%3A%22P1+for+MVP%22

## Design
- [ ] TODO for @lpatmo: follow up with folks in #pair-programming on whether we missed any missing details
- [ ] We went over early Figma designs in a Twitch stream with @adachiu, who shared thoughtful questions/comments, and @BethanyG summed up the 💡 moment re: our new hangouts direction in a spot-on comment:

![image](https://user-images.githubusercontent.com/4512699/91800460-db6ee880-ebdd-11ea-8e1a-aa7191088e05.png)
- [ ] @adachiu is doing some thinking on a new home page design for CBV3
- [ ] IMPORTANT in progress: user stories threads at https://github.com/codebuddies/frontend/discussions/138. Important discussion for hangouts! 

## Priorities before we can deploy
Thanks to @billglover for offering to help with deployment! 

Probably some things we want to get done before we deploy:
- [ ] For the FE, finish up create/edit
- [ ] For the BE, finish up email verification for new accounts
- [ ] For design, finalize the new hangout flow user story / show it to a couple of potential new users

## Twitch/YouTube
**Goal: work on CBV3 in public, and showcase cool open source projects/learnings from members of the community!** 

**Show & Tell**
- [x] @billglover shared about his side project, a chinese reader browser extension, at https://www.youtube.com/watch?v=SkEUs99idx4
- [ ] @TerrabyteTiger will be exploring GitHub Actions on Thursday at 8:15pm Eastern! RSVP here: https://codebuddies.org/hangout/ziaCimXr36FQWHAr8
- [x] Signup google spreadsheet: https://docs.google.com/spreadsheets/u/2/d/1kNhAR1nf5CaD-amJ3fJ1K4nKK5hBmx1GpTYDWmls2e4/edit#gid=0

**Miscellaneous**
- [x] Improvements on the video quality in Twitch streams -- @lpatmo figured out how to stream directly to Twitch from Zoom (h/t @billglover for sharing the [tutorial](https://tech.paulcz.net/blog/streaming-from-zoom-to-twitch/) on it) and is working on figuring out how to make this easy for alternate hosts. 
- [x] @TerrabyteTiger designed awesome cover art for his upcoming GitHub Actions stream (https://twitter.com/TerabyteTiger/status/1300397934043975680) and @lpatmo recreated the design in Google Slides: https://docs.google.com/presentation/d/1M4KkA3WLI1hWR2m3FKiCRzDTMkv7aKfSNV6HmsCaF3c/edit#slide=id.p so that we can replicate new slides for future speakers easily. 
![image](https://user-images.githubusercontent.com/4512699/91801706-5ab0ec00-ebdf-11ea-9837-d56945c584fd.png)
![image](https://user-images.githubusercontent.com/4512699/91801916-69979e80-ebdf-11ea-8bbc-bffa6b0fce63.png)

- [ ] Hopefully more CBV3 "watch us as we work on PRs" streams soon

## Sponsors
- Special thanks to github.com/gitduck for sponsoring us ($1000) in July! You can join CodeBuddies on GitDuck using this special invite link: https://gitduck.com/codebuddies/join?t=60ktFkh1Rqnd_AS1kR8ZGyH
- Special thanks to DigitalOcean for renewing the project with $600 in credits!
- [ ] Plan to write about CB architecture using DigitalOcean after deployment. Notes from @billglover: 

![image](https://user-images.githubusercontent.com/4512699/91800399-c1350a80-ebdd-11ea-83d5-44e7b830fc42.png)

## Newsletter
No idea when the next update is 😂  but maybe some of these updates could go in there

## Slack Community
Special thanks to @stain88 for keeping a close eye on channitor, archiving a bunch of channels, and for an amazing spreadsheet documenting all the 100+ channels (and joining every single channel)

Thanks to all the moderators in #ask-in-admin for helping out where you've been able to! 🙏 

**TODO:**
- [ ] Look into whether we should install a timezone converter like https://channeltime.info/ or https://spacetime.am/pricing. h/t @BethanyG for those recs. This convo was sparked after a member shared a personal project about a time converter he was working on that's not open-sourced yet, but am looking forward to hearing about when it is!

## Study Groups
- A couple of people have mentioned on Slack (or DMed me) about study groups they want to organize to go over a tutorial or work on a project... wondering about the best way to help support and amplify 📣  those groups, and whether to continue to recommend that they be created in codebuddies.org/study-groups where unfortunately it's hard to figure out which ones are still active. 
- Otherwise, a bunch of people seem to be self-organizing on Slack/meeting privately, which is great!

**Comment by angelocordon** [comment]
What a lovely update - thanks so much! Great work team, thanks to all the
contributors, admins and administrators!

**Comment by lpatmo** [comment]
September 6th small weekend update:
![image](https://user-images.githubusercontent.com/4512699/92338472-86135b00-f065-11ea-8bd0-53d34666026a.png)
